qt/scientific_interfaces/Indirect/IndexTypes.h

```diff
@@ -23,7 +23,7 @@ namespace IDA {
  * the same type.
  */
 template <int Class> struct IndexType {
-  using IntImplementationType = int;
+  using IntImplementationType = size_t;
```

Comment:
I think this has the advantage of making it clear which functions depend on exactly what index type in being wrapped and which do not. For example it is not otherwise obvious at least to me that when you change the internal type to also need to change some types in the IndexCollection types below. 

More long term I am hoping to remove these types entirely and consolidate them with the existing types in IndexType.